commit b7ed754e6615d35e329eebba646ad7abc211ee76 (refs/changes/67/967/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-11-29T08:09:28-05:00 (2 years, 10 months ago)
    
    joyent/node-cueball#53 CueBallDNSResolver#resolve fails to handle CNAME to a non-existent name

diff --git a/lib/resolver.js b/lib/resolver.js
index aa0474f..3f1e625 100644
--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -924,6 +924,12 @@ CueBallDNSResolver.prototype.resolve = function (domain, type, timeout) {
 		} else {
 			throw (new Error('Invalid record type ' + type));
 		}
+
+		if (ans.length === 0) {
+			err = new NoRecordsError(domain, type);
+			em.emit('error', err);
+			return;
+		}
 		em.emit('answers', ans, minTTL);
 	}
 };
diff --git a/package.json b/package.json
index dde05e3..4e94367 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "1.1.6",
+  "version": "1.1.7",
   "description": "",
   "main": "lib/index.js",
   "dependencies": {
